src/core/smp.cc

```diff
@@ -173,11 +179,32 @@ smp::setup_prefaulter(const seastar::resource::resources& res, seastar::memory::
 #endif
 }
 
+static
+std::optional<size_t>
+get_huge_page_size() {
+    auto meminfo_fd = file_desc::open("/proc/meminfo", O_RDONLY | O_CLOEXEC);
+    std::string meminfo;
+    char buf[4096];
+    while (auto size_opt = meminfo_fd.read(buf, sizeof(buf))) {
+        if (!*size_opt) {
+            break;
+        }
+        meminfo.append(buf, *size_opt);
+    }
+    static std::regex re(R"(Hugepagesize:\s*(\d+) kB)");
```

Comment:
Nitpick: I don't think the outer parentheses are needed.

By the way, I hope nobody will ever change the kernel to print any other units except kB in /proc/meminfo :-) I believe it's hard-coded to kB right now, so it's fine.

Reply:
The outer parentheses are needed. The middle parentheses are part of the `R"` literal syntax.

Reply:
Wow, I never even heard of this crazy syntax, and now it turns out it already existed in C++11. This language is a disaster :-(